src/mtype.d

```diff
@@ -7963,13 +7948,14 @@ public:
                 return oe;
             return new DotExp(e.loc, e, oe);
         }
+
         Declaration d = s.isDeclaration();
-        debug
+        if (!d)
         {
-            if (!d)
-                printf("d = %s '%s'\n", s.kind(), s.toChars());
+            e.error("%s.%s is not a declaration", e.toChars(), ident.toChars());
```

Comment:
I understand, but the same case is handled by `error` in `TypeClass.dotExp`. I couldn't find a related test case for the error report in `TypeClass.dotExp`, so I concluded that this line would have possibility to cause assertion failure in released compiler, because whena symbol can be declared in class members, equivalent symbol can also be declared in struct member.

Honestly, similar error report exists at the end of `DsymbolExp.resolve` function (`"%s '%s' is not a variable"`). In old days I thought it would be unreachable and could be replaced with `assert`, but it was actually reached because of other place bug (see [issue 14858](https://issues.dlang.org/show_bug.cgi?id=14858)). I think this replacement is better so it's more defensive.